src/commands/runDraftTool/runDraftWorkflow.ts

```diff
@@ -0,0 +1,561 @@
+   AzureWizard,
+   AzureWizardExecuteStep,
+   AzureWizardPromptStep,
+   IActionContext
+} from '@microsoft/vscode-azext-utils';
+import {ignoreFocusOut} from './helper/commonPrompts';
+import {buildGenerateWorkflowCommand} from './helper/draftCommandBuilder';
+import {longRunning} from '../../utils/host';
+import {
+   Az,
+   AzApi,
+   getAzCreds,
+   ManagedClusterItem,
+   RegistryItem,
+   RepositoryItem,
+   ResourceGroupItem,
+   SubscriptionItem
+} from '../../utils/az';
+import {getAsyncResult} from '../../utils/errorable';
+import {sort} from '../../utils/sort';
+import {getBranches} from '../../utils/gitExtension';
+import {Branch, Ref} from '../../utils/git';
+import path = require('path');
+
+const title = 'Draft a GitHub Actions Workflow';
+
+interface PromptContext {
+   destination: vscode.Uri;
+   clusterSubscription: SubscriptionItem;
+   clusterResourceGroup: ResourceGroupItem;
+   cluster: ManagedClusterItem;
+   acrSubscription: SubscriptionItem;
+   acrResourceGroup: ResourceGroupItem;
+   registry: RegistryItem;
+   newRepository: boolean;
+   acrRepository: RepositoryItem;
+   branch: string;
+   selectRepoDir: boolean;
+}
+type WizardContext = IActionContext & Partial<PromptContext>;
+type IPromptStep = AzureWizardPromptStep<WizardContext>;
+type IExecuteStep = AzureWizardExecuteStep<WizardContext>;
+
+export async function runDraftWorkflow({
+   actionContext,
+   extensionContext
+}: Context) {
+   const az: AzApi = new Az(getAzCreds);
+
+   // Ensure Draft Binary
+   const downloadResult = await longRunning(`Downloading Draft.`, () =>
+      ensureDraftBinary()
+   );
+   if (!downloadResult) {
+      vscode.window.showErrorMessage('Failed to download Draft');
+      return undefined;
+   }
+
+   if (
+      !vscode.workspace.workspaceFolders ||
+      vscode.workspace.workspaceFolders.length === 0
+   ) {
+      throw Error('No directories within current vscode wokspace.');
+   }
+
+   const promptSteps: IPromptStep[] = [
+      new PromptAKSSubscriptionSelection(az),
+      new PromptAKSResourceGroupSelection(az),
+      new PromptAKSClusterSelection(az),
+      // Currently we dont allow seperate sub for ACR selection while generating through draft
+      // new PromptACRSubscriptionSelection(az),
+      new PromptACRResourceGroupSelection(az),
+      new PromptACRSelection(az),
+      new PromptACRRegistrySelection(az),
+      new PromptNewRepository(),
+      new PromptGitHubBranchSelection()
+   ];
+   const executeSteps: IExecuteStep[] = [
+      new ExecuteDraftWorkflow(),
+      new ExecuteOpenWorkflowFile()
+   ];
+
+   const wizardContext: WizardContext = {
+      ...actionContext,
+      destination: vscode.workspace.workspaceFolders[0].uri
+   };
+
+   const wizard = new AzureWizard(wizardContext, {
+      title,
+      promptSteps,
+      executeSteps
+   });
+   await wizard.prompt();
+   await wizard.execute();
+}
+class PromptAKSSubscriptionSelection extends AzureWizardPromptStep<WizardContext> {
```

Comment:
https://github.com/Azure/aks-devx-tools/blob/main/src/commands/runDraftTool/helper/commonPrompts.ts

In general we should be using common prompts for subscription and rg.

Reply:
done